mailserver/mailserver_test.go

```diff
@@ -191,25 +192,60 @@ func (s *MailserverSuite) TestSetupRequestMessageDecryptor() {
 	config = *s.config
 	s.NoError(config.ReadMailServerPasswordFile())
 	s.NoError(s.server.Init(s.shh, &config))
-	s.NotNil(s.server.filter.KeySym)
-	s.Nil(s.server.filter.KeyAsym)
+	s.Require().NotNil(s.server.symFilter)
+	s.NotNil(s.server.symFilter.KeySym)
+	s.Nil(s.server.asymFilter)
 	s.server.Close()
 
 	// AsymKey can also be used
 	config = *s.config
 	s.NoError(config.ReadMailServerAsymKeyFile())
 	s.NoError(s.server.Init(s.shh, &config))
-	s.Nil(s.server.filter.KeySym) // important: symmetric key should be nil
-	s.Equal(config.MailServerAsymKey, s.server.filter.KeyAsym)
+	s.Nil(s.server.symFilter) // important: symmetric filter should be nil
+	s.Require().NotNil(s.server.asymFilter)
+	s.Equal(config.MailServerAsymKey, s.server.asymFilter.KeyAsym)
 	s.server.Close()
 
-	// when both Password and AsymKey are set, Password has a preference
+	// when Password and AsymKey are set, both are supprted
 	config = *s.config
 	s.NoError(config.ReadMailServerPasswordFile())
 	s.NoError(config.ReadMailServerAsymKeyFile())
 	s.NoError(s.server.Init(s.shh, &config))
-	s.NotNil(s.server.filter.KeySym)
-	s.Nil(s.server.filter.KeyAsym)
+	s.Require().NotNil(s.server.symFilter)
+	s.NotNil(s.server.symFilter.KeySym)
+	s.NotNil(s.server.asymFilter.KeyAsym)
+	s.server.Close()
+}
+
+func (s *MailserverSuite) TestOpenEnvelope() {
+	// Password should work ok
+	config := *s.config
+	s.NoError(config.ReadMailServerPasswordFile())
+	s.NoError(s.server.Init(s.shh, &config))
+	s.Require().NotNil(s.server.symFilter)
```

Comment:
I think you can skip checking `s.server.symFilter` and others here. Just generate an envelope and call `openEnvelope`.

Reply:
This specific assertion Is required, as in case `s.server.symFilter` is nil, trying to access its public key to sign the envelope will produce a panic.
```
	symKey := s.server.symFilter.KeySym
	env, err := generateEnvelopeWithKeys(time.Now(), symKey, nil)
```
I'll remove some of the non needed assertions

Reply:
I mean that this test function should only verify `openEnvelope` method as its name says. Instead, it verifies a lot of other things. These other things are already checked in a previous test function.

Reply:
@adambabik I've split the test on 2 different functions and added some comments hoping it looks more clear. 
Could you be more precise on what's not required on this test? As it's basically:
- setting up a server with a sym/asym key
- preparing an envelope with the same key as the server
- testing openEnvelope

Do you mean I should have something like `config.ReadMailServerPasswordFile()` instead of `s.NoError(config.ReadMailServerPasswordFile())`

```
func (s *MailserverSuite) TestOpenEnvelopeWithSymKey() {
	// Setup the server with a sym key
	config := *s.config
	s.NoError(config.ReadMailServerPasswordFile())
	s.NoError(s.server.Init(s.shh, &config))

	// Prepare a valid envelope
	s.Require().NotNil(s.server.symFilter)
	symKey := s.server.symFilter.KeySym
	env, err := generateEnvelopeWithKeys(time.Now(), symKey, nil)
	s.Require().NoError(err)

	// Test openEnvelope with a valid envelope
	d := s.server.openEnvelope(env)
	s.NotNil(d)
	s.Equal(testPayload, d.Payload)
	s.server.Close()
}
```